### PR TITLE
Fixed the bug that causes PPO performance gap

### LINKED ISSUE #1059: PPO cartpole unittest fails sometimes
Two issues:
1) the unittest is not deterministic, different runs may produce different results. One example is that the ppo_cart_pole test fails sometimes.
2) verify if there is still a performance gap after #1055

### PR DESCRIPTION
**Pull request by Haichao-Zhang**
This PR fixed a bug introduced in  #1055 that causes the behavior difference for PPO .

### CODE REVIEW

**Comment by hnyu** [comment]
So you mean cache (the more accurate result) actually led to worse results in our test case?

File: alf/utils/dist_utils.py [hunk 1]
```diff
@@ -853,7 +853,9 @@ def _compute_log_prob(single_distribution, single_action):
     return total_log_probs
 
 
-def rsample_action_distribution(nested_distributions, return_log_prob=False):
+def rsample_action_distribution(nested_distributions,
+                                return_log_prob=False,
+                                disable_cache=True):
```
Comment (emailweixu) [inline alf/utils/dist_utils.py]: I feel we should default it to False. And config it to True for ppo_cart_pole.

**Comment by Haichao-Zhang** [comment]
> So you mean cache (the more accurate result) actually led to worse results in our test case?

Still under testing. Will push more updates.

File: alf/utils/dist_utils.py (hunk 1 above)
Comment (Haichao-Zhang) [reply alf/utils/dist_utils.py]: > I feel we should default it to False. And config it to True for ppo_cart_pole.

Will do. 
Current test shows that even if we disable the cache, the ppo_cartpole test can still fail, which is strange. 
I'm still doing more testing on this.

File: alf/algorithms/ppo_algorithm.py [hunk 2]
```diff
@@ -52,7 +52,6 @@ def train_step(self, inputs: TimeStep, state, rollout_info):
                 reward=alg_step.info.reward,
                 discount=alg_step.info.discount,
                 action_distribution=alg_step.info.action_distribution,
-                rollout_log_prob=alg_step.info.log_prob,
```
Comment (Haichao-Zhang) [inline alf/algorithms/ppo_algorithm.py:55]: Introduced a bug here in the previous change. The ``rollout_info.rollout_log_prob`` are extracted from replay buffer by the ``preprocess_experience`` function, and should not be modified here.

**Review by hnyu — APPROVED** [review]

File: alf/utils/dist_utils.py (hunk 1 above)
Comment (Haichao-Zhang) [reply alf/utils/dist_utils.py]: Found the root cause and pushed a new commit fixing the bug.

**Review by breakds — APPROVED** [review]

**Review by emailweixu — APPROVED** [review]

**Pull request merged**